shared/entities/base-data.view.ts

```diff
@@ -0,0 +1,214 @@
+import { ViewColumn, ViewEntity } from "typeorm";
+
+@ViewEntity({
```

Comment:
Nice, congrats in this ultra massive join xD

I am wondering how will we manage the update/refresh of the view? I don't see this already happening here but I might be missing smth.

If not, do you have already some idea as to how to do it? Should we have some listener/trigger that performs a refresh in the view for each insert/update on some targeted tables?